common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExecutorDiskUtils.java

```diff
@@ -27,7 +27,7 @@
 
 public class ExecutorDiskUtils {
 
-  private static final Pattern MULTIPLE_SEPARATORS = Pattern.compile(File.separator + "{2,}");
+  private static final Pattern MULTIPLE_SEPARATORS = Pattern.compile("[/\\\\]+");
```

Comment:
Can we add another test case where there are three file separators?

Reply:
@attilapiros was this simply a mistake or intentionally used 2 to match? Seems the current change is correct but wanted make doubly sure.

Reply:
I think the case already covered by https://github.com/apache/spark/pull/28940/files#diff-e48223108a52336484e4bf803d62221bR159-R160

Reply:
As before implement, assume all path sep is `/`, only over 2 `/` should be replaced by `/`. But now, consider the path `/a\b/\c` on Windows, the single `/` also should be replaced

Reply:
Ah, okay. I read it as `{2}` and I just noticed it's actually `{2,}`. Can we compile the pattern differently after checking the OS?  Looks like it's going to replace `\\` in the file name on other Linux base OSs.

Reply:
You are right, `\\` is legal char in filename on Unix-like OS, but not on Windows, use different pattern is a better choice.

Reply:
Hm, that one would probably need to fix too. I sort of audited and managed to make all tests pass on Windows .. but seems like there are a bunch of new instances to fix again ..

Reply:
I have a sort of half automated script to run the tests on Windows via CI via AppVeyor which I used at that time to audit the tests on Windows. Let me try to run and see if that works:

Build started: [CORE] `org.apache.spark.network.shuffle.ExternalShuffleBlockResolverSuite` [![PR-28940](https://ci.appveyor.com/api/projects/status/github/spark-test/spark?branch=6BE4E4C8-09CE-4166-85DD-53C91C022221&svg=true)](https://ci.appveyor.com/project/spark-test/spark/branch/6BE4E4C8-09CE-4166-85DD-53C91C022221)

Reply:
Ok .. at least it seems started to build.. :-). I will run it again tomorrow against the latest commit here.